src/Channels.Networking.TLS/SecurityContext.cs

```diff
@@ -0,0 +1,198 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Security.Authentication;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Channels.Networking.TLS.Internal;
+
+namespace Channels.Networking.TLS
+{
+    public class SecurityContext: IDisposable
+    {
+        internal const ContextFlags RequiredFlags = ContextFlags.ReplayDetect | ContextFlags.SequenceDetect | ContextFlags.Confidentiality | ContextFlags.AllocateMemory;
+        internal const ContextFlags ServerRequiredFlags = RequiredFlags | ContextFlags.AcceptStream;
+        internal const int BlockSize = 1024 * 4 - 64; //Current fixed block size
+        private const string SecurityPackage = "Microsoft Unified Security Protocol Provider";
+        public const int MaxStackAllocSize = 32 * 1024;
```

Comment:
Being low in the stack, remember this is unwound quickly and is sync so requires no copying/preserving of the stack